### ISSUE TITLE
ceph mgr crashes continuously when using rook 1.12 or 1.13 with ceph 18.2

### ISSUE THREAD

**Issue by mnlipp** [rook/rook#13380]
* Bug Report

**Deviation from expected behavior:**

After enabling the display of "cluster utulization" in the dashboard with  `ceph dashboard set-prometheus-api-host http://prometheus.monitoring:9090`, ceph mgr crashes continuously,. info is:
```
{
    "archived": "2023-12-12 21:38:42.265720",
    "backtrace": [
        "  File \"/usr/share/ceph/mgr/orchestrator/_interface.py\", line 675, in get_prometheus_access_info\n    raise NotImplementedError()",
        "NotImplementedError"
    ],
    "ceph_version": "18.2.0",
    "crash_id": "2023-12-12T21:16:15.532426Z_bede3cde-c3ba-41f4-afab-9522e582b8b1",
    "entity_name": "mgr.b",
    "mgr_module": "rook",
    "mgr_module_caller": "ActivePyModule::dispatch_remote get_prometheus_access_info",
    "mgr_python_exception": "NotImplementedError",
    "os_id": "centos",
    "os_name": "CentOS Stream",
    "os_version": "8",
    "os_version_id": "8",
    "process_name": "ceph-mgr",
    "stack_sig": "bbf52dcdbbe54d67edf59ebdb5d201fffd921db5a9dd4431964c2aaac2250c7e",
    "timestamp": "2023-12-12T21:16:15.532426Z",
    "utsname_hostname": "tc1-h3",
    "utsname_machine": "x86_64",
    "utsname_release": "5.14.0-284.30.1.el9_2.x86_64",
    "utsname_sysname": "Linux",
    "utsname_version": "#1 SMP PREEMPT_DYNAMIC Tue Sep 12 09:28:32 EDT 2023"
}
```

**Expected behavior:**

Should not crash

**Comment by travisn** [comment]
Rook v1.12 or greater is required for Ceph v18. Previous versions of Rook are not supported with Reef. But that's actually quite amazing If you've been able to upgrade Ceph for so long without upgrading Rook.

**Comment by mnlipp** [comment]
> Rook v1.12 or greater is required for Ceph v18. Previous versions of Rook are not supported with Reef. But that's actually quite amazing If you've been able to upgrade Ceph for so long without upgrading Rook.

Please excuse the typo.

**Comment by travisn** [comment]
> `prometheus-api-host http://prometheus.monitoring:9090`, ceph mgr crashes continuously,. info is:

Phew, ok. :)

It's possible this is already fixed in v18.2.1 that is due out any day now. To test a pre-release image, if you have a test cluster could you see if this repros with this image? `quay.io/ceph/daemon-base:latest-reef-devel`. Or else try again when v18.2.1 comes out.

If it's still an issue, please do open a [Ceph tracker issue](https://tracker.ceph.com/) as this would need to be fixed on the ceph side.

**Comment by mnlipp** [comment]
Are you sure that this is a ceph problem? I think the cause is that the rook orch backend does not support ´orch prometheus info` (that's in the logs). Is the support for this by a backend optonal?

**Comment by travisn** [comment]
The rook orch backend is implemented in the ceph repo as a mgr module, so any failure in the mgr modules needs to go through the ceph repo.

**Comment by mnlipp** [comment]
(This is currently pending because of https://tracker.ceph.com/issues/63868. I'm not sure if I only have no data in prometheus or something else is not working properly.)

**Comment by mnlipp** [comment]
I can see the metrics captured with v18.2.0 after upgrading to v18.2.1 without crashes (I just don't get new metrics). So this issue is solved by v18.2.1.
